### DISCUSSION TITLE
error with nginx when changing ssl cert

### DISCUSSION THREAD

**Discussion by byronsims** [Security-Onion-Solutions/securityonion#7793] [Unsupported Versions]
soversion:  2.3.100
Deployed:  On-Prem
OS:  CentOS 7

I followed the directions in: https://github.com/Security-Onion-Solutions/securityonion/discussions/5117

I am not using Let's Encrypt, but certs are from a commercial CA.

The issue I'm having is nginx does not seem to start.  If I do an so-status it says "MISSING" next to nginx. 
The /opt/so/log/nginx/error.log shows:
"2022/04/14 17:50:08 [emerg] 1#1: SSL_CTX_use_PrivateKey("/etc/pki/nginx/server.key") failed (SSL: error:0B080074:x509 certificate routines:X509_check_private_key:key values mismatch)"

My managerssl.crt file has 4 certs listed, the root, intermediate 1, intermediate 2, and the cert for the box.  In that order.  The managerssl.key file is vaild for the cert for the box in the managerssl.crt.  I have also tried flipping the certs in managerssl.crt the other direction to see if I had that backwards.

Do you know what I need to do to?

Thanks,
Byron

**Comment by byronsimseku** [comment]
Any ideas on this one?

**Comment by dougburks** [comment]
I would suggest asking the search engine of your choice about the error message `X509_check_private_key:key values mismatch` and looking at the results, specifically:
https://www.namecheap.com/support/knowledgebase/article.aspx/9781/2238/nginx-ssl-error0b080074x509-certificate-routines-x509_check_private_keykey-values-mismatch/

**Comment by tomjones1977** [comment]
Are you sure it's the correct private key file?  The error looks like the public cert and private key don't match.  I'd double check the content of those files.

**Comment by elenezet** [comment]
Hi,

I have the same problem on a server, did you find a solution?
Thanks

**Reply by esher72** [reply]
I recently changed mine and had this happen in my test lab. My cert file had the entire cert chain in it. I pulled out JUST the cert for the console and used that with the private key I used to create it. Rinse, lather, repeat in prod with no issues.

**Reply by elenezet** [reply]
i use openssl x509 command on the server after that it work fine to me.
